builtInFunctions/esdtMetaDataUpdate.go

```diff
@@ -0,0 +1,161 @@
+	"math/big"
+	"sync"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
+)
+
+type esdtMetaDataUpdate struct {
+	baseActiveHandler
+	funcGasCost           uint64
+	globalSettingsHandler vmcommon.GlobalMetadataHandler
+	storageHandler        vmcommon.ESDTNFTStorageHandler
+	rolesHandler          vmcommon.ESDTRoleHandler
+	accounts              vmcommon.AccountsAdapter
+	enableEpochsHandler   vmcommon.EnableEpochsHandler
+	gasConfig             vmcommon.BaseOperationCost
+	mutExecution          sync.RWMutex
+}
+
+// NewESDTMetaDataUpdateFunc returns the esdt meta data update built-in function component
+func NewESDTMetaDataUpdateFunc(
+	funcGasCost uint64,
+	gasConfig vmcommon.BaseOperationCost,
+	accounts vmcommon.AccountsAdapter,
+	globalSettingsHandler vmcommon.GlobalMetadataHandler,
+	storageHandler vmcommon.ESDTNFTStorageHandler,
+	rolesHandler vmcommon.ESDTRoleHandler,
+	enableEpochsHandler vmcommon.EnableEpochsHandler,
+) (*esdtMetaDataUpdate, error) {
+	if check.IfNil(accounts) {
+		return nil, ErrNilAccountsAdapter
+	}
+	if check.IfNil(globalSettingsHandler) {
+		return nil, ErrNilGlobalSettingsHandler
+	}
+	if check.IfNil(enableEpochsHandler) {
+		return nil, ErrNilEnableEpochsHandler
+	}
+	if check.IfNil(storageHandler) {
+		return nil, ErrNilESDTNFTStorageHandler
+	}
+	if check.IfNil(rolesHandler) {
+		return nil, ErrNilRolesHandler
+	}
+
+	e := &esdtMetaDataUpdate{
+		accounts:              accounts,
+		globalSettingsHandler: globalSettingsHandler,
+		storageHandler:        storageHandler,
+		rolesHandler:          rolesHandler,
+		enableEpochsHandler:   enableEpochsHandler,
+		funcGasCost:           funcGasCost,
+		gasConfig:             gasConfig,
+		mutExecution:          sync.RWMutex{},
+	}
+
+	e.baseActiveHandler.activeHandler = enableEpochsHandler.IsDynamicESDTEnabled
+
+	return e, nil
+}
+
+// ProcessBuiltinFunction saves the token type in the system account
+func (e *esdtMetaDataUpdate) ProcessBuiltinFunction(acntSnd, _ vmcommon.UserAccountHandler, vmInput *vmcommon.ContractCallInput) (*vmcommon.VMOutput, error) {
+	err := checkDynamicArguments(vmInput, acntSnd, e.baseActiveHandler, 7, e.rolesHandler, core.ESDTRoleNFTUpdate)
+	if err != nil {
+		return nil, err
+	}
+
+	totalLengthDifference := 0
+	for _, arg := range vmInput.Arguments {
+		totalLengthDifference += len(arg)
+	}
+
+	esdtInfo, err := getEsdtInfo(vmInput, acntSnd, e.storageHandler, e.globalSettingsHandler)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(vmInput.Arguments[nameIndex]) != 0 {
+		totalLengthDifference -= len(esdtInfo.esdtData.TokenMetaData.Name)
+		esdtInfo.esdtData.TokenMetaData.Name = vmInput.Arguments[nameIndex]
+
+	}
+	totalLengthDifference -= len(esdtInfo.esdtData.TokenMetaData.Creator)
+	esdtInfo.esdtData.TokenMetaData.Creator = vmInput.CallerAddr
+
+	if len(vmInput.Arguments[royaltiesIndex]) != 0 {
+		totalLengthDifference -= 32
```

Comment:
why 32 ? make the substraction with len(vmInput.Arguments[royaltiesIndex])

Reply:
Done.